dll/win32/imm32/precomp.h

```diff
@@ -0,0 +1,95 @@
+/*
+ * PROJECT:     ReactOS IMM32
+ * LICENSE:     LGPL-2.1-or-later (https://spdx.org/licenses/LGPL-2.1-or-later)
+ * PURPOSE:     Implementing IMM32
+ * COPYRIGHT:   Copyright 1998 Patrik Stridvall
+ *              Copyright 2002, 2003, 2007 CodeWeavers, Aric Stewart
+ *              Copyright 2017 James Tabor <james.tabor@reactos.org>
+ *              Copyright 2018 Amine Khaldi <amine.khaldi@reactos.org>
+ *              Copyright 2020 Oleg Dubinskiy <oleg.dubinskij2013@yandex.ua>
+ *              Copyright 2020-2021 Katayama Hirofumi MZ <katayama.hirofumi.mz@gmail.com>
+ */
+
+#pragma once
+
+#include <stdarg.h>
+#include <stdio.h>
+
+#define WIN32_NO_STATUS
+#include <windef.h>
+#include <winbase.h>
+#include <wingdi.h>
+#include <winuser.h>
+#include <winerror.h>
+#include <wine/debug.h>
+#include <imm.h>
+#include <ddk/imm.h>
+#include <winnls.h>
+#include <winreg.h>
+#include <wine/list.h>
+#include <stdlib.h>
+#include <ndk/umtypes.h>
+#include <ndk/pstypes.h>
+#include <ndk/rtlfuncs.h>
+#include "ntuser.h"
+#include "ntwin32.h"
+#include <undocuser.h>
+#include <imm32_undoc.h>
+#include <strsafe.h>
```

Comment:
```suggestion
#include <stdarg.h>
#include <stdio.h>
#include <stdlib.h>

/* PSDK/NDK Headers */
#define WIN32_NO_STATUS
#include <windef.h>
#include <winbase.h>
#include <winerror.h>
#include <wingdi.h>
#include <winuser.h>
#include <winnls.h>
#include <winreg.h>

#include <imm.h>
#include <ddk/imm.h>

#define NTOS_MODE_USER
#include <ndk/umtypes.h>
#include <ndk/pstypes.h>
#include <ndk/rtlfuncs.h>

/* Public Win32K Headers */
#include "ntuser.h"
#include "ntwin32.h"

/* Undocumented user definitions */
#include <undocuser.h>
#include <imm32_undoc.h>

#include <strsafe.h>

#include <wine/debug.h>
#include <wine/list.h>
```

Reply:
Are `winerror.h` and `wine/list.h` still useful?

Reply:
`<winerror.h>` is removed. `<wine/list.h>` is still used from legacy code.